clean_old_aggregates.py

```diff
@@ -0,0 +1,70 @@
+import keystoneauth1
+import datetime
+from keystoneauth1.identity import v3
+from keystoneauth1 import loading
+from keystoneauth1 import session
+from novaclient import client as nova_client
+from blazarclient import client as blazar_client
+
+def main():
+
+    loader = loading.get_plugin_loader('password')
+    auth = loader.load_from_options(auth_url='http://REDACTED:35357',
+    username='admin',
+    password='REDACTED',
+    project_id='REDACTED',
+    user_domain_id ='default')
+
+    sess = session.Session(auth=auth)
+
+    nova = nova_client.Client(version='2', region_name='CHI@TACC', session=sess)
+    blazar = blazar_client.Client(1, session=sess, service_type='reservation', region_name='CHI@TACC')
+
+    term_leases=[]
+    agglist=[]
+    aggdetails=[]
+    aggnames=[]
+    exp_aggs=[]
+    agg_host=[]
+
+    # Find leases with end date before current time
+    leases=blazar.lease.list()
+
+    time=datetime.datetime.now()
+    for x in leases:
+        lease_time=datetime.datetime.strptime(x['end_date'], '%Y-%m-%dT%H:%M:%S.%f')
```

Comment:
Watch out for timezones here. You should add a `.replace(tzinfo=timezone('UTC'))` at the end here to tell Python this is a UTC timestamp. That should make the time comparison more reliable. I think `.now()` will use the local time zone by default, not sure. I would use `.utcnow()` instead in any case, so you're comparing apples to apples.